src/components/ChannelList/hooks/useChannelVisibleListener.js

```diff
@@ -0,0 +1,36 @@
+// @ts-check
+
+import { useEffect, useContext } from 'react';
+import uniqBy from 'lodash.uniqby';
+import { ChatContext } from '../../../context';
+import { getChannel } from '../utils';
+
+/**
+ * @typedef {import('stream-chat').Event<string>} ChannelVisibleEvent
+ * @typedef {React.Dispatch<React.SetStateAction<import('stream-chat').Channel[]>>} SetChannels
+ * @param {SetChannels} setChannels
+ * @param {(setChannels: SetChannels, event: ChannelVisibleEvent) => void} [customHandler]
+ */
+export const useChannelVisibleListener = (setChannels, customHandler) => {
+  const { client } = useContext(ChatContext);
+  useEffect(() => {
+    /** @param {import('stream-chat').Event<string>} e */
+    const handleEvent = async (e) => {
+      console.log('channel.visible event', e);
+
+      if (customHandler && typeof customHandler === 'function') {
+        customHandler(setChannels, e);
+      } else if (e?.type) {
+        const channel = await getChannel(client, e.channel_type, e.channel_id);
+        setChannels((channels) => uniqBy([channel, ...channels], 'cid'));
```

Comment:
shouldn't this part run even if there are no custom handler?

Reply:
in other functions we its an override, so not an additional function to be called.